### ISSUE TITLE
Improve logging interface

### ISSUE THREAD

**Issue by bowenwang1996** [nearprotocol/near-runtime-ts#126]
Currently the function available function for logging is `log` which takes a string. It would be better if we can support logging of struct without having to call `.encode` explicitly. I am not sure how exactly this can be done given the constraint of assemblyscript (there is no macro for example), but it's something that we should keep in mind.

**Comment by willemneal** [comment]
Yeah this can be done easily with a generic function.  [See here](https://github.com/jtenner/as-pect/blob/5db6c1464bd5ea98340704fbbb6ec61cefb9280c/packages/assembly/assembly/internal/log.ts#L53).

**Comment by bowenwang1996** [comment]
That helps but is still not ideal. Ideally you want something like rust's `println` macro that allows you to easily mix message with structs.

**Comment by willemneal** [comment]
Do you mean like string interpolation?

**Comment by bowenwang1996** [comment]
Yes

**Comment by willemneal** [comment]
Ah well then @ailisp and I have been working on that separately.  We can easily add a default `toString` method for each reference type to get its JSON string.  So for now I say that we can provide the `toString` and then users will still have to use concatenation while we finish up adding interpolation to upstream.

**Comment by bowenwang1996** [comment]
That's good enough for now. The problem right now is that people are not aware of the `.encode` method so they cannot log an object easily.

**Comment by willemneal** [comment]
Yeah we need to add a section of setting up the typing environment.  @amgando and I can meet about it in the near future, (hehe near ;-) ).  It the types file [here](https://github.com/nearprotocol/near-runtime-ts/blob/master/assembly/as_types.d.ts) I was able to  inject the methods into any reference type.  However, if you don't add this file to your local setup through the use of `/// <refercence types="near-runtime-ts/assemblys/as_types" />` then the IDE doesn't hint that this method is possible.
